src/main/java/com/wix/mysql/EmbeddedMysql.java

```diff
@@ -14,13 +14,15 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.ReentrantLock;
 
 import static com.wix.mysql.config.MysqldConfig.SystemDefaults.SCHEMA;
 import static com.wix.mysql.utils.Utils.or;
 import static java.lang.String.format;
 
 public class EmbeddedMysql {
     private final static Logger logger = LoggerFactory.getLogger(EmbeddedMysql.class);
+    private static final ReentrantLock localRepository = new ReentrantLock();
```

Comment:
@viliusl this will only work when running it from the same process, why not sync download process using a lock file ?